extensions/amp-app-banner/amp-app-banner.md

```diff
@@ -0,0 +1,158 @@
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# <a name="amp-app-banner"></a> `amp-app-banner`
+
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>A wrapper and minimal UI for a cross-platform, fixed-position banner showing a call-to-action to install an app. Includes conditional logic to direct to the right app on the right platform, and to hide permenantly if the user dismisses the banner.</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td>Experimental</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Required Script</strong></td>
+    <td>
+      <div>
+        <code>&lt;script async custom-element="amp-app-banner" src="https://cdn.ampproject.org/v0/amp-app-banner-0.1.js">&lt;/script></code>
+      </div>
+    </td>
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong><a href="https://www.ampproject.org/docs/guides/responsive/control_layout.html">Supported Layouts</a></strong></td>
+    <td>nodisplay</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td>
+      <ul>
+        <li>
+          <a
+          href="https://amp-mk-3.herokuapp.com/examples/article.amp.max.html">article.amp.max.html (outside of viewer context)</a>
+        </li>
+        <li>
+          <a href="https://amp-mk-3.herokuapp.com/examples/viewer.html">viewer.html (inside of viewer context — tap "Two" at the top of the page to show the page containing the banner)</a>
+        </li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+## Behavior
+`amp-app-banner` is a wrapper and minimal UI for a cross-platform, fixed-position banner showing a call-to-action to install an app. Includes conditional logic to direct to the right app on the right platform, and to hide permenantly if the user dismisses the banner.
+
+
+##Data Sources
+To extend and promote the usage of the natively supported app banners, we are using the exact data-sources the native app banners use on the respective platforms. iOS uses a `<meta name="apple-itunes-app">` tag in the head of the document and Android uses a `<link rel="manifest">`. 
+
+The AMP runtime parses the meta tag content attribute on iOS extracting the App ID and `app-argument` (usually used for deep-linking URIs - app-protocols like `whatsapp://` or `medium://`). On Android, the AMP runtimes makes an XHR request to fetch the `manifest.json` file, and parses its content to extract `app_id` from `related_applications` and it calculates the app store URL as well as open-in-app URL:
+
+```
+android-app://${appId}/${protocol}/${pathname}
```

Comment:
`android-app://${appId}/${protocol}/${host}${pathname}`

Note that the protocol/host/pathname are calculated from the canonical URL of the AMP document and that your native app need to register the links in their manifest as [documented here](https://developer.android.com/training/app-indexing/deep-linking.html).

I'd also suggest maybe including the app manifest snippet in here.